src/main/java/utils/storage/json/JsonStorage.java

```diff
@@ -69,146 +58,43 @@ public JsonStorage(String filePath) {
         } catch (IOException e) {
-
             String absolutePath = this.saveFile.getAbsolutePath();
             logger.log(Level.WARNING, "Failed to load file from " + absolutePath, e);
-
             throw new StorageLoadFailure(absolutePath);
         } catch (NullPointerException | JsonSyntaxException e) {
             String absolutePath = this.saveFile.getAbsolutePath();
             logger.log(Level.WARNING, "Corrupted save file: " + absolutePath, e);
-            //useBackup = true;
-            throw new StorageCorrupted(absolutePath);
+            return loadBackup();
         }
+    }
 
-        if (useBackup == true) {
-            logger.log(Level.INFO, "Trying to load backup file");
-            try {
-                FileReader fileReader = new FileReader(backupFile);
-                BufferedReader bufferedReader = new BufferedReader(fileReader);
-                JsonElement jsonElement = gsonBuilder.create().fromJson(bufferedReader, JsonElement.class);
-                JsonObject jsonObject = jsonElement.getAsJsonObject();
-                String deckName = jsonObject.get("deckName").getAsString();
-
-                JsonArray jsonArray = jsonObject.getAsJsonArray("cards");
-                Type cardListType = new TypeToken<ArrayList<Card>>() {
-                }.getType();
-
-                ArrayList<Card> cards = gsonBuilder.create().fromJson(jsonArray, cardListType);
-                cardList = new CardList(cards);
-            } catch (IOException | NullPointerException | JsonSyntaxException ex) {
-                String absolutePath = this.backupFile.getAbsolutePath();
-                logger.log(Level.WARNING, "Corrupted backup file: " + absolutePath, ex);
-                throw new StorageCorrupted(absolutePath);
-            }
+    private Memory loadBackup() throws InkaException {
+        try {
+            FileReader fileReader = new FileReader(backupFile);
+            BufferedReader bufferedReader = new BufferedReader(fileReader);
+            JsonElement jsonElement = gsonBuilder.create().fromJson(bufferedReader, JsonElement.class);
+            JsonObject saveDataObject = jsonElement.getAsJsonObject();
+            Memory savedMemory = JsonToMemory.convert(saveDataObject);
+            logger.log(Level.INFO, "Loaded backup file successfully");
+            return savedMemory;
+        } catch (IOException | NullPointerException | JsonSyntaxException ex) {
+            String absolutePath = this.backupFile.getAbsolutePath();
+            logger.log(Level.WARNING, "Corrupted backup file: " + absolutePath, ex);
+            throw new StorageCorrupted(absolutePath);
         }
+    }
 
-        Memory savedMemory = new Memory();
-        savedMemory.setCardList(cardList);
-        savedMemory.setTagList(tagList);
-        savedMemory.setDeckList(deckList);
 
-        return savedMemory;
-    }
 
     public static Memory makeMemory(CardList cardList, TagList tagList, DeckList deckList) {
```

Comment:
I think you can remove this since `Memory(cardList, tagList, deckList)` does the same thing. Also, I'm in favour of removing the `setCardList/setTagList/setDeckList` methods in case we accidentally overwrite them somewhere in the code

Reply:
it's in case we want to modify memory to be continuously modified throughout the app's runtime